data-avro/src/main/java/com/linkedin/data/avro/DataTranslator.java

```diff
@@ -666,6 +645,19 @@ else if (fieldValue == null)
                 continue;
               }
             }
+
+            if ((fieldDataSchema.getDereferencedType() != DataSchema.Type.UNION) && fieldAvroSchema.getType() == Schema.Type.UNION) {
```

Comment:
nit: remove unnecessary parentheses

Reply:
I just think it hard to read. I split this line into two so I can remove unnecessary parentheses but still make it look nice.